src/main.rs

```diff
@@ -313,7 +313,23 @@ fn construct_config(mut opts: Opts, pattern_regexps: &[String]) -> Result<Config
             .map(crate::fmt::FormatTemplate::parse),
         command: command.map(Arc::new),
         batch_size: opts.batch_size,
-        exclude_patterns: opts.exclude.iter().map(|p| String::from("!") + p).collect(),
+        exclude_patterns: opts
+            .exclude
+            .iter()
+            .map(|p| {
+                let path = Path::new(p);
+                let relative_path = if path.is_absolute() {
+                    let cwd = env::current_dir().unwrap_or("".into());
```

Comment:
This won't work if one or more search paths is given to fd, since it needs to be relative to the search path, which may not be the current directory.